### PR TITLE
[Merged by Bors] - docs: reorganized usage guide and 'implementation.adoc'

### PR DESCRIPTION
**Pull request by fhennig**
# Description

- Split up usage into multiple pages
- deleted implementation.adoc
  - some content was moved to the index.adoc
- fixed a link that used md syntax
- removed a nightly reference that was marked with TODO
- added more cross-links so readers find things easier

## Review Checklist

- [ ] Code contains useful comments
- [ ] CRD change approved (or not applicable)
- [ ] (Integration-)Test cases added (or not applicable)
- [ ] Documentation added (or not applicable)
- [ ] Changelog updated (or not applicable)
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: docs/modules/ROOT/pages/index.adoc [hunk 1]
```diff
@@ -1,18 +1,42 @@
 = Stackable Operator for Apache HDFS
 
-This is an operator for Kubernetes that can manage https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-hdfs/HdfsUserGuide.html[Apache HDFS] clusters.
+The Stackable Operator for https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-hdfs/HdfsUserGuide.html[Apache HDFS] is used to set up HFDS in high-availability mode. It depends on the xref:zookeeper:ROOT:index.adoc[] to operate a ZooKeeper cluster to coordinate the active/passive name node.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/index.adoc]:
```suggestion
The Stackable Operator for https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-hdfs/HdfsUserGuide.html[Apache HDFS] is used to set up HFDS in high-availability mode. It depends on the xref:zookeeper:ROOT:index.adoc[] to operate a ZooKeeper cluster to coordinate the active and standby name nodes.
```

File: docs/modules/ROOT/pages/index.adoc [hunk 2]
```diff
@@ -1,18 +1,42 @@
...
 
-WARNING: This operator only works with images from the https://repo.stackable.tech/#browse/browse:docker:v2%2Fstackable%2Fhadoop[Stackable] repository
+NOTE: This operator only works with images from the https://repo.stackable.tech/#browse/browse:docker:v2%2Fstackable%2Fhadoop[Stackable] repository
+
+== Roles
+
+Three xref:home:concepts:roles-and-role-groups.adoc[roles] of the HDFS cluster are implemented:
+
+* Data Node - responsible for holding the actual data. *IMPORTANT* Currently the data is kept in `hostPath` volumes of 1GB and no configuration options are exposed to the user. Each data node has its own volume.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/index.adoc]: I dont understand. The user can configure (recently with https://github.com/stackabletech/hdfs-operator/pull/291 even multiple) volumes for datanodes.
```suggestion
* Data Node - responsible for storing the actual data. *IMPORTANT* Currently the data is kept in `hostPath` volumes of 1GB and no configuration options are exposed to the user. Each data node has its own volume.
```

File: docs/modules/ROOT/pages/index.adoc [hunk 3]
```diff
@@ -1,18 +1,42 @@
...
+* Journal Node - responsible for keeping track of HDFS blocks and used to perform failovers in case the active name node fails. For details see: https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-hdfs/HDFSHighAvailabilityWithQJM.html
+* Name Node - responsible for keeping track of HDFS blocks and providing access to the data.
+
+== Kubernetes objects
+
+The operator creates the following K8S objects per role group defined in the custom resource.
+
+* ClusterIP - used for intra-cluster communication.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/index.adoc]: The K8s object is a `Service`
```suggestion
* Service - ClusterIP used for intra-cluster communication.
```

File: docs/modules/ROOT/pages/index.adoc [hunk 4]
```diff
@@ -1,18 +1,42 @@
...
+* ConfigMap - HDFS configuration files like `core-site.xml` and `hdfs-site.xml` are defined here and mounted in the pods.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/index.adoc]: log4j.properties config as well.

File: docs/modules/ROOT/pages/index.adoc [hunk 5]
```diff
@@ -1,18 +1,42 @@
...
+* StatefulSet - where the replica count of each role group is defined. By default, a cluster will have 2 name nodes, 3 journal nodes and 3 data nodes.
+
+In addition, a `NodePort` service is created for each pod that exposes all container ports to the outside world (from the perspective of K8S).
```
Comment (maltesander) [inline docs/modules/ROOT/pages/index.adoc]:
```suggestion
In addition, a `NodePort` service is created for each pod labeled with `hdfs.stackable.tech/pod-service=true` that exposes all container ports to the outside world (from the perspective of K8S).
```

File: docs/modules/ROOT/pages/index.adoc [hunk 6]
```diff
@@ -1,18 +1,42 @@
...
+* Journal Node - responsible for keeping track of HDFS blocks and used to perform failovers in case the active name node fails. For details see: https://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-hdfs/HDFSHighAvailabilityWithQJM.html
+* Name Node - responsible for keeping track of HDFS blocks and providing access to the data.
+
+== Kubernetes objects
+
+The operator creates the following K8S objects per role group defined in the custom resource.
+
+* ClusterIP - used for intra-cluster communication.
+* ConfigMap - HDFS configuration files like `core-site.xml` and `hdfs-site.xml` are defined here and mounted in the pods.
+* StatefulSet - where the replica count of each role group is defined. By default, a cluster will have 2 name nodes, 3 journal nodes and 3 data nodes.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/index.adoc]: Not sure this is true anymore.

File: docs/modules/ROOT/pages/usage-guide/monitoring.adoc [hunk 7]
```diff
@@ -0,0 +1,9 @@
+= Monitoring
+
+The cluster can be monitored with Prometheus from inside or outside the K8S cluster.
+
+All services (with the exception of the Zookeeper daemon on the node names) run with the JMX exporter agent enabled and expose metrics on the `metrics` port. This port is available form the container level up to the NodePort services.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/usage-guide/monitoring.adoc]:
```suggestion
All services (with the exception of the Zookeeper daemon on the node names) run with the JMX exporter agent enabled and expose metrics on the `metrics` port. This port is available from the container level up to the NodePort services.
```

**Review by maltesander — CHANGES_REQUESTED** [review]
Minor things..

File: docs/modules/ROOT/pages/index.adoc (hunk 2 above)
Comment (fhennig) [reply docs/modules/ROOT/pages/index.adoc]: Maybe that is also outdated then? Do you think the comment should be removed? Maybe we can link to docs that talk about setting up the volumes?

Comment (fhennig) [reply docs/modules/ROOT/pages/index.adoc]: I have removed the statement [de6a8bd](https://github.com/stackabletech/hdfs-operator/pull/302/commits/de6a8bd9d50ff2fd3445a522faf79360f0b16409)

File: docs/modules/ROOT/pages/index.adoc (hunk 4 above)
Comment (fhennig) [reply docs/modules/ROOT/pages/index.adoc]: [9957d54](https://github.com/stackabletech/hdfs-operator/pull/302/commits/9957d54d6b97e2ba7aa1f0f571e0ce3dbf5475b6)

File: docs/modules/ROOT/pages/index.adoc (hunk 6 above)
Comment (fhennig) [reply docs/modules/ROOT/pages/index.adoc]: [c2c24bc](https://github.com/stackabletech/hdfs-operator/pull/302/commits/c2c24bcc0b18a53e022f0208165446e371e5bb34)

**Review by maltesander — APPROVED** [review]
LGTM.

**Comment by fhennig** [comment]
bors merge

**Pull request closed without merging**